### PR TITLE
Two more Packable derive

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

**Review by Thoralf-M — APPROVED** [review]

File: sdk/src/types/block/core/block.rs [hunk 1]
```diff
@@ -111,60 +115,29 @@ impl BlockHeader {
...
-        Ok(Self {
-            protocol_version,
-            network_id,
-            issuing_time,
-            slot_commitment_id,
-            latest_finalized_slot,
-            issuer_id,
-        })
+fn verify_network_id<const VERIFY: bool>(network_id: &u64, params: &ProtocolParameters) -> Result<(), Error> {
+    if VERIFY && *network_id != params.network_id() {
```
Comment (Alex6323) [inline sdk/src/types/block/core/block.rs:133]:
```suggestion
    if VERIFY && network_id != &params.network_id() {
```
not that it mattered, but I would prefer this.

File: sdk/src/types/block/core/block.rs [hunk 2]
```diff
@@ -111,60 +115,29 @@ impl BlockHeader {
...
-        self.slot_commitment_id.pack(packer)?;
-        self.latest_finalized_slot.pack(packer)?;
-        self.issuer_id.pack(packer)?;
-
-        Ok(())
+fn verify_protocol_version<const VERIFY: bool>(
+    protocol_version: &u8,
+    params: &ProtocolParameters,
+) -> Result<(), Error> {
+    if VERIFY && *protocol_version != params.version() {
```
Comment (Alex6323) [inline sdk/src/types/block/core/block.rs:122]:
```suggestion
    if VERIFY && protocol_version != &params.version() {
```
my preference

File: sdk/src/types/block/core/block.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/types/block/core/block.rs:133]: We mostly dereference in verify functions so that would be inconsistent

Comment (Alex6323) [reply sdk/src/types/block/core/block.rs:133]: hmm, okay

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**